### PR TITLE
[office-js][office-js-preview] (Outlook) Document error message when legacy Exchange tokens are turned off

### PR DESCRIPTION
**Pull request by samantharamon**
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:

- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: None.
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

File: types/office-js-preview/index.d.ts [hunk 1]
```diff
@@ -17746,11 +17746,17 @@ declare namespace Office {
...
          *
-         * - `InternalServerError`: The Exchange server returned an error. Please look at the diagnostics object for more information.
+         * - `HTTPRequestFailure: The request has failed. Please look at the diagnostics object for the HTTP error code`.
          *
-         * - `NetworkError`: The user is no longer connected to the network. Please check your network connection and try again.
+         * - `InternalServerError: The Exchange server returned an error. Please look at the diagnostics object for more information`. - In Exchange Online environments,
+         * this error occurs when the token can't be retrieved because legacy Exchange tokens for Outlook add-ins are turned off. We recommend using NAA as a single sign-on solution for your add-in.
+         * For guidance on how to implement NAA, see
+         * {@link https://learn.microsoft.com/office/dev/add-ins/develop/enable-nested-app-authentication-in-your-add-in | Enable SSO in an Office Add-in using nested app authentication}.
+         * - `NetworkError: The user is no longer connected to the network. Please check your network connection and try again.`
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts:17759]: Need an extra line here?

```suggestion
         *
         * - `NetworkError: The user is no longer connected to the network. Please check your network connection and try again.`
```

File: types/office-js/index.d.ts [hunk 2]
```diff
@@ -15817,11 +15831,18 @@ declare namespace Office {
          *
          * **Errors**:
          *
-         * - `HTTPRequestFailure`: The request has failed. Please look at the diagnostics object for the HTTP error code.
+         * - `GenericTokenError: An internal error has occurred.` - In Exchange Online environments, this error occurs when the token can't be retrieved because legacy Exchange tokens
+         * for Outlook add-ins are turned off. We recommend using NAA as a single sign-on solution for your add-in. For guidance on how to implement NAA, see
+         * {@link https://learn.microsoft.com/office/dev/add-ins/develop/enable-nested-app-authentication-in-your-add-in | Enable SSO in an Office Add-in using nested app authentication}.
+         *
+         * - `HTTPRequestFailure: The request has failed. Please look at the diagnostics object for the HTTP error code`.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts:15838]: I noticed some of these include the period within the backticks while others don't. Is it that the error message does/doesn't end with a period?

File: types/office-js-preview/index.d.ts [hunk 3]
```diff
@@ -17746,11 +17746,17 @@ declare namespace Office {
          *
          * **Errors**:
          *
-         * - `HTTPRequestFailure`: The request has failed. Please look at the diagnostics object for the HTTP error code.
+         * - `GenericTokenError: An internal error has occurred.` - In Exchange Online environments, this error occurs when the token can't be retrieved because legacy Exchange tokens
+         * for Outlook add-ins are turned off. We recommend using NAA as a single sign-on solution for your add-in. For guidance on how to implement NAA, see
+         * {@link https://learn.microsoft.com/office/dev/add-ins/develop/enable-nested-app-authentication-in-your-add-in | Enable SSO in an Office Add-in using nested app authentication}.
```
Comment (davidchesnut) [inline types/office-js-preview/index.d.ts:17751]: Again I think we should link to the FAQ https://aka.ms/naafaq as that will have more comprehensive information about deprecation of the tokens, or how to turn them on/off if someone wants to do that.

File: types/office-js-preview/index.d.ts [hunk 4]
```diff
@@ -17746,11 +17746,17 @@ declare namespace Office {
...
          *
-         * - `InternalServerError`: The Exchange server returned an error. Please look at the diagnostics object for more information.
+         * - `HTTPRequestFailure: The request has failed. Please look at the diagnostics object for the HTTP error code`.
          *
-         * - `NetworkError`: The user is no longer connected to the network. Please check your network connection and try again.
+         * - `InternalServerError: The Exchange server returned an error. Please look at the diagnostics object for more information`. - In Exchange Online environments,
+         * this error occurs when the token can't be retrieved because legacy Exchange tokens for Outlook add-ins are turned off. We recommend using NAA as a single sign-on solution for your add-in.
+         * For guidance on how to implement NAA, see
+         * {@link https://learn.microsoft.com/office/dev/add-ins/develop/enable-nested-app-authentication-in-your-add-in | Enable SSO in an Office Add-in using nested app authentication}.
```
Comment (davidchesnut) [inline types/office-js-preview/index.d.ts:17758]: link to FAQ please.

File: types/office-js-preview/index.d.ts [hunk 5]
```diff
@@ -17746,11 +17746,17 @@ declare namespace Office {
          *
          * **Errors**:
          *
-         * - `HTTPRequestFailure`: The request has failed. Please look at the diagnostics object for the HTTP error code.
+         * - `GenericTokenError: An internal error has occurred.` - In Exchange Online environments, this error occurs when the token can't be retrieved because legacy Exchange tokens
+         * for Outlook add-ins are turned off. We recommend using NAA as a single sign-on solution for your add-in. For guidance on how to implement NAA, see
+         * {@link https://learn.microsoft.com/office/dev/add-ins/develop/enable-nested-app-authentication-in-your-add-in | Enable SSO in an Office Add-in using nested app authentication}.
          *
-         * - `InternalServerError`: The Exchange server returned an error. Please look at the diagnostics object for more information.
+         * - `HTTPRequestFailure: The request has failed. Please look at the diagnostics object for the HTTP error code`.
```
Comment (davidchesnut) [inline types/office-js-preview/index.d.ts:17753]: We have multiple diagnostics objects and it took me a while to figure out this was actually directly on the result object. Might be nice just to link it to that specific version of the object. The other ones don't have the additional JSON error info. https://learn.microsoft.com/en-us/javascript/api/office/office.asyncresult?#office-office-asyncresult-diagnostics-member

**Review by davidchesnut — CHANGES_REQUESTED** [review]
Looks good but with a couple suggestions.

File: types/office-js-preview/index.d.ts [hunk 6]
```diff
@@ -17164,11 +17164,11 @@ declare namespace Office {
...
+         * - In February 2025, legacy Exchange {@link https://learn.microsoft.com/office/dev/add-ins/outlook/authentication#exchange-user-identity-token | user identity} and
          * {@link https://learn.microsoft.com/office/dev/add-ins/outlook/authentication#callback-tokens | callback} tokens will be turned off by default for all Exchange Online tenants.
          * This is part of {@link https://blogs.microsoft.com/on-the-issues/2023/11/02/secure-future-initiative-sfi-cybersecurity-cyberattacks/ | Microsoft's Secure Future Initiative},
          * which gives organizations the tools needed to respond to the current threat landscape. Exchange user identity tokens will still work for Exchange on-premises.
-         * Nested app authentication is the recommended approach for tokens going forward. For more information, see our {@link https://aka.ms/NAApreviewblog | blog post} and
+         * Nested app authentication (NAA) is the recommended approach for tokens going forward. For more information, see our {@link https://aka.ms/NAApreviewblog | blog post} and
```
[the remark this answers is not in the document]
Comment (samantharamon) [reply types/office-js-preview/index.d.ts:17171]: Good callout. Changed to FAQ link. Here and in other instances.

File: types/office-js-preview/index.d.ts (hunk 3 above)
Comment (samantharamon) [reply types/office-js-preview/index.d.ts:17751]: Updated.

File: types/office-js-preview/index.d.ts (hunk 5 above)
Comment (samantharamon) [reply types/office-js-preview/index.d.ts:17753]: That's a great suggestion. Added a link.

File: types/office-js/index.d.ts (hunk 2 above)
Comment (samantharamon) [reply types/office-js/index.d.ts:15838]: The period should be within the backticks. Fixed the typo.

File: types/office-js-preview/index.d.ts [hunk 7]
```diff
@@ -17164,12 +17164,11 @@ declare namespace Office {
...
+         * - In February 2025, legacy Exchange {@link https://learn.microsoft.com/office/dev/add-ins/outlook/authentication#exchange-user-identity-token | user identity} and
          * {@link https://learn.microsoft.com/office/dev/add-ins/outlook/authentication#callback-tokens | callback} tokens will be turned off by default for all Exchange Online tenants.
          * This is part of {@link https://blogs.microsoft.com/on-the-issues/2023/11/02/secure-future-initiative-sfi-cybersecurity-cyberattacks/ | Microsoft's Secure Future Initiative},
          * which gives organizations the tools needed to respond to the current threat landscape. Exchange user identity tokens will still work for Exchange on-premises.
-         * Nested app authentication is the recommended approach for tokens going forward. For more information, see our {@link https://aka.ms/NAApreviewblog | blog post} and
-         * {@link https://learn.microsoft.com/office/dev/add-ins/outlook/faq-nested-app-auth-outlook-legacy-tokens | FAQ page}.
+         * Nested app authentication (NAA) is the recommended approach for tokens going forward. For more information, see our {@link https://aka.ms/naafaq | FAQ page}.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts:17171]: Here and other instances...

```suggestion
         * Nested app authentication (NAA) is the recommended approach for tokens going forward. For more information, see the {@link https://aka.ms/naafaq | FAQ page}.
```

File: types/office-js-preview/index.d.ts [hunk 8]
```diff
@@ -17681,12 +17680,11 @@ declare namespace Office {
...
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts:17687]:
```suggestion
         * Nested app authentication (NAA) is the recommended approach for tokens going forward. For more information, see the {@link https://aka.ms/naafaq | FAQ page}.
```

File: types/office-js-preview/index.d.ts [hunk 9]
```diff
@@ -17777,12 +17784,11 @@ declare namespace Office {
...
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts:17791]:
```suggestion
         * Nested app authentication (NAA) is the recommended approach for tokens going forward. For more information, see the {@link https://aka.ms/naafaq | FAQ page}.
```

File: types/office-js-preview/index.d.ts [hunk 10]
```diff
@@ -17943,12 +17958,11 @@ declare namespace Office {
...
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts:17965]:
```suggestion
         * Nested app authentication (NAA) is the recommended approach for tokens going forward. For more information, see the {@link https://aka.ms/naafaq | FAQ page}.
```

File: types/office-js-preview/index.d.ts [hunk 11]
```diff
@@ -18054,12 +18077,11 @@ declare namespace Office {
...
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts:18084]:
```suggestion
         * Nested app authentication (NAA) is the recommended approach for tokens going forward. For more information, see the {@link https://aka.ms/naafaq | FAQ page}.
```

File: types/office-js/index.d.ts [hunk 12]
```diff
@@ -15541,12 +15540,11 @@ declare namespace Office {
...
+         * - In February 2025, legacy Exchange {@link https://learn.microsoft.com/office/dev/add-ins/outlook/authentication#exchange-user-identity-token | user identity} and
          * {@link https://learn.microsoft.com/office/dev/add-ins/outlook/authentication#callback-tokens | callback} tokens will be turned off by default for all Exchange Online tenants.
          * This is part of {@link https://blogs.microsoft.com/on-the-issues/2023/11/02/secure-future-initiative-sfi-cybersecurity-cyberattacks/ | Microsoft's Secure Future Initiative},
          * which gives organizations the tools needed to respond to the current threat landscape. Exchange user identity tokens will still work for Exchange on-premises.
-         * Nested app authentication is the recommended approach for tokens going forward. For more information, see our {@link https://aka.ms/NAApreviewblog | blog post} and
-         * {@link https://learn.microsoft.com/office/dev/add-ins/outlook/faq-nested-app-auth-outlook-legacy-tokens | FAQ page}.
+         * Nested app authentication (NAA) is the recommended approach for tokens going forward. For more information, see our {@link https://aka.ms/naafaq | FAQ page}.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts:15547]: Here and other instances...

```suggestion
         * Nested app authentication (NAA) is the recommended approach for tokens going forward. For more information, see the {@link https://aka.ms/naafaq | FAQ page}.
```

**Review by davidchesnut — APPROVED** [review]
Looks great!

**Review by ElizabethSamuel-MSFT — APPROVED** [review]

**Review by Rick-Kirkham — APPROVED** [review]

**Pull request merged**